### ISSUE TITLE
Traceback issue

### ISSUE THREAD

**Issue by dc52241** [jeffreydwalter/arlo#59]
Please answer these questions before submitting your issue. Thanks!

### What version of Python are you using (`python -V`)?
Python 3.6.4 :: Anaconda, Inc.

### Which Python packages do you have installed (run the `pip freeze` or `pip3 freeze` command and paste output)?alabaster==0.7.10
anaconda-client==1.6.9
anaconda-navigator==1.7.0
anaconda-project==0.8.2
arlo==1.0.9
asn1crypto==0.24.0
astroid==1.6.1
astropy==2.0.3
attrs==17.4.0
Babel==2.5.3
backports.shutil-get-terminal-size==1.
beautifulsoup4==4.6.0
bitarray==0.8.1
bkcharts==0.2
blaze==0.11.3
bleach==2.1.2
bokeh==0.12.13
boto==2.48.0
Bottleneck==1.2.1
certifi==2018.1.18
cffi==1.11.4
chardet==3.0.4
click==6.7
cloudpickle==0.5.2
clyent==1.2.2
colorama==0.3.9
comtypes==1.1.4
conda==4.4.10
conda-build==3.4.1
conda-verify==2.0.0
contextlib2==0.5.5
cryptography==2.1.4
cycler==0.10.0
Cython==0.27.3
cytoolz==0.9.0
dask==0.16.1
datashape==0.5.4
decorator==4.2.1
distributed==1.20.2
docutils==0.14
entrypoints==0.2.3
et-xmlfile==1.0.1
fastcache==1.0.2
filelock==2.0.13
Flask==0.12.2
Flask-Cors==3.0.3
gevent==1.2.2
glob2==0.6
greenlet==0.4.12
h5py==2.7.1
heapdict==1.0.0
html5lib==1.0.1
idna==2.6
imageio==2.2.0
imagesize==0.7.1
ipython==6.2.1
ipython-genutils==0.2.0
ipywidgets==7.1.1
isort==4.2.15
itsdangerous==0.24
jdcal==1.3
jedi==0.11.1
Jinja2==2.10
jsonschema==2.6.0
jupyter==1.0.0
jupyter-client==5.2.2
jupyter-console==5.2.0
jupyter-core==4.4.0
jupyterlab==0.31.4
jupyterlab-launcher==0.10.2
lazy-object-proxy==1.3.1
llvmlite==0.21.0
locket==0.2.0
lxml==4.1.1
MarkupSafe==1.0
matplotlib==2.1.2
mccabe==0.6.1
menuinst==1.4.11
mistune==0.8.3
monotonic==1.4
mpmath==1.0.0
msgpack-python==0.5.1
multipledispatch==0.4.9
navigator-updater==0.1.0
nbconvert==5.3.1
nbformat==4.4.0
networkx==2.1
nltk==3.2.5
nose==1.3.7
notebook==5.4.0
numba==0.36.2
numexpr==2.6.4
numpy==1.14.0
numpydoc==0.7.0
odo==0.5.1
olefile==0.45.1
openpyxl==2.4.10
packaging==16.8
pandas==0.22.0
pandocfilters==1.4.2
parso==0.1.1
partd==0.3.8
path.py==10.5
pathlib2==2.3.0
patsy==0.5.0
pep8==1.7.1
pickleshare==0.7.4
Pillow==5.0.0
pkginfo==1.4.1
pluggy==0.6.0
ply==3.10
prompt-toolkit==1.0.15
psutil==5.4.3
py==1.5.2
pycodestyle==2.3.1
pycosat==0.6.3
pycparser==2.18
pycrypto==2.6.1
pycurl==7.43.0.1
pyflakes==1.6.0
Pygments==2.2.0
pylint==1.8.2
pyodbc==4.0.22
pyOpenSSL==17.5.0
pyparsing==2.2.0
PySocks==1.6.7
pytest==3.3.2
python-dateutil==2.6.1
pytz==2017.3
PyWavelets==0.5.2
pywin32==222
pywinpty==0.5
PyYAML==3.12
pyzmq==16.0.3
QtAwesome==0.4.4
qtconsole==4.3.1
QtPy==1.3.1
requests==2.18.4
rope==0.10.7
ruamel-yaml==0.15.35
scikit-image==0.13.1
scikit-learn==0.19.1
scipy==1.0.0
seaborn==0.8.1
Send2Trash==1.4.2
simplegeneric==0.8.1
singledispatch==3.4.0.3
six==1.11.0
snowballstemmer==1.2.1
sortedcollections==0.5.3
sortedcontainers==1.5.9
Sphinx==1.6.6
sphinxcontrib-websupport==1.0.1
spyder==3.2.6
SQLAlchemy==1.2.1
sseclient==0.0.19
statsmodels==0.8.0
sympy==1.1.1
tables==3.4.2
tblib==1.3.2
terminado==0.8.1
testpath==0.3.1
toolz==0.9.0
tornado==4.5.3
traitlets==4.3.2
typing==3.6.2
unicodecsv==0.14.1
urllib3==1.22
wcwidth==0.1.7
webencodings==0.5.1
Werkzeug==0.14.1
widgetsnbextension==3.1.0
win-inet-pton==1.0.1
win-unicode-console==0.5
wincertstore==0.2
wrapt==1.10.11
xlrd==1.1.0
XlsxWriter==1.0.2
xlwings==0.11.5
xlwt==1.3.0
zict==0.1.3

### Which Arlo hardware do you have (camera types - [Arlo, Pro, Q, etc.], basestation model, etc.)?
Arlo Go

### What did you do?
I had made two scripts, one that captures an image and one to capture a video. Both of these worked fine and now when I run them I get an error. I can run manually in spyder line by line, but if I just run with python and script name I get errors. 

If possible, provide the steps you took to reproduce the issue. 
A complete runnable program is good. (don't include your user/password or any sensitive info)
from Arlo import Arlo
import time
snapshot.py
USERNAME = 'XXXXXXXXXXXXX'
PASSWORD = 'XXXXXXXXXXXX'

try:
    timestr = time.strftime("%Y%m%d-%H%M%S")
    # Instantiating the Arlo object automatically calls Login(), which returns an oAuth token that gets cached.
    # Subsequent successful calls to login will update the oAuth token.
    arlo = Arlo(USERNAME, PASSWORD)
    # At this point you're logged into Arlo.
    # Get the list of devices and filter on device type to only get the camera.
    # This will return an array which includes all of the camera's associated metadata.
    basestations = arlo.GetDevices('basestation')
    cameras = arlo.GetDevices('camera')
    # Tells the Arlo basestation to trigger a snapshot on the given camera.
    # This snapshot is not instantaneous, so this method waits for the response and returns the url
    # for the snapshot, which is stored on the Amazon AWS servers. 
    snapshot_url = arlo.TriggerFullFrameSnapshot(cameras[0], cameras[0]) 
    # This method requests the snapshot for the given url and writes the image data to the location specified.
    # In this case, to the current directory as a file named "snapshot.jpg"
    # Note: Snapshots are in .jpg format.  timestr = time.strftime("%Y%m%d-%H%M%")
    arlo.DownloadSnapshot(snapshot_url, 'JPG/05454300 '+timestr+'.jpg')

except Exception as e:
    print(e)   

### What did you see instead?
Traceback (most recent call last):
  File "<pyshell#10>", line 1, in <module>
    snapshot_url = arlo.TriggerFullFrameSnapshot(cameras[0], cameras[0])
  File "C:\Python36-32\lib\site-packages\Arlo.py", line 810, in TriggerFullFrameSnapshot
    return self.TriggerAndHandleEvent(basestation, trigger, callback)
  File "C:\Python36-32\lib\site-packages\Arlo.py", line 445, in TriggerAndHandleEvent
    self.Subscribe(basestation)
  File "C:\Python36-32\lib\site-packages\Arlo.py", line 299, in Subscribe
    self.event_streams[basestation_id].Start()
  File "C:\Python36-32\lib\site-packages\Arlo.py", line 85, in Start
    self.event_stream_thread.start()
AttributeError: 'EventStream' object has no attribute 'event_stream_thread'

### Does this issue reproduce with the latest release?
Yes

**Comment by jeffreydwalter** [comment]
Hi, can you please remove the  #try/except block, starting on line 50 and try your code again. I think you're probably having an issue connecting to the EventStream and the exception is being ignored.

https://github.com/jeffreydwalter/arlo/blob/master/Arlo.py#L50-L55

**Comment by dc52241** [comment]
I removed, I still cannot run from cmd, I get the following below. If I run in spyder though it works fine. Hmm.

 File "C:\Users\dechrist\arlo\snapshot.py", line 20, in <module>
   snapshot_url = arlo.TriggerFullFrameSnapshot(cameras[0], cameras[0])
 File "C:\Python36-32\lib\site-packages\Arlo.py", line 810, in TriggerFullFrame
napshot
   return self.TriggerAndHandleEvent(basestation, trigger, callback)
 File "C:\Python36-32\lib\site-packages\Arlo.py", line 445, in TriggerAndHandle
vent
   self.Subscribe(basestation)
 File "C:\Python36-32\lib\site-packages\Arlo.py", line 299, in Subscribe
   self.event_streams[basestation_id].Start()
 File "C:\Python36-32\lib\site-packages\Arlo.py", line 85, in Start
   self.event_stream_thread.start()
ttributeError: 'EventStream' object has no attribute 'event_stream_thread'
xception ignored in: <bound method EventStream.__del__ of <Arlo.EventStream obj
ct at 0x02FE4F30>>
raceback (most recent call last):
 File "C:\Python36-32\lib\site-packages\Arlo.py", line 58, in __del__
 File "C:\Python36-32\lib\site-packages\Arlo.py", line 93, in Disconnect
 File "C:\Python36-32\lib\site-packages\Arlo.py", line 110, in Unregister
ttributeError: 'EventStream' object has no attribute 'event_stream_thread'

**Comment by jeffreydwalter** [comment]
Did you just comment out the try/except lines? It doesn't look like your line numbers in the trackback have changed.

**Comment by jeffreydwalter** [comment]
Actually, it looks like you don't have the latest version of the `arlo` library. Can you please upgrade to the latest and try again.

**Comment by dc52241** [comment]
So I updated to the newest version of Arlo arlo==1.0.10. Updated code below. This code was working for weeks with no problem, from a batch file. I changed one part of one line, and it seems to have stopped working "'JPG/05454300 '", directory name change and file prefix. This code works when I run it from Spyder, but no longer when I just run from cmd or a batch file.  

**The updated code below.**
from Arlo import Arlo
import time

USERNAME = XXXXXXXX
PASSWORD = XXXXXXXX

timestr = time.strftime("%Y%m%d-%H%M%S")
arlo = Arlo(USERNAME, PASSWORD)
basestations = arlo.GetDevices('basestation')
cameras = arlo.GetDevices('camera')
snapshot_url = arlo.TriggerFullFrameSnapshot(cameras[0], cameras[0])
arlo.DownloadSnapshot(snapshot_url, 'JPG/05454300 '+timestr+'.jpg')
